uber/models/department.py

```diff
@@ -0,0 +1,389 @@
+                    'DeptMembership.is_dept_head == True)',
+        secondary='dept_membership',
+        order_by='Attendee.full_name',
+        viewonly=True)
+    checklist_admins = relationship(
+        'Attendee',
+        backref=backref('checklist_depts', order_by='Department.name'),
+        cascade='save-update,merge,refresh-expire,expunge',
+        primaryjoin='and_('
+                    'Department.id == DeptMembership.department_id, '
+                    'DeptMembership.is_checklist_admin == True)',
+        secondary='dept_membership',
+        order_by='Attendee.full_name',
+        viewonly=True)
+    pocs = relationship(
+        'Attendee',
+        backref=backref('poc_depts', order_by='Department.name'),
+        cascade='save-update,merge,refresh-expire,expunge',
+        primaryjoin='and_('
+                    'Department.id == DeptMembership.department_id, '
+                    'DeptMembership.is_poc == True)',
+        secondary='dept_membership',
+        order_by='Attendee.full_name',
+        viewonly=True)
+    members = relationship(
+        'Attendee',
+        backref=backref('assigned_depts', order_by='Department.name'),
+        cascade='save-update,merge,refresh-expire,expunge',
+        order_by='Attendee.full_name',
+        secondary='dept_membership')
+    memberships = relationship('DeptMembership', backref='department')
+    explicit_membership_requests = relationship(
+        'DeptMembershipRequest', backref='department')
+    attendees_requesting_membership = relationship(
+        'Attendee',
+        backref=backref('requested_depts', order_by='Department.name'),
+        cascade='save-update,merge,refresh-expire,expunge',
+        primaryjoin='or_('
+                    'DeptMembershipRequest.department_id == Department.id, '
+                    'DeptMembershipRequest.department_id == None)',
+        secondary='dept_membership_request',
+        order_by='Attendee.full_name',
+        viewonly=True)
+    parent = relationship(
+        'Department',
+        backref=backref(
+            'sub_depts',
+            order_by='Department.name',
+            cascade='all,delete-orphan'),
+        cascade='save-update,merge,refresh-expire,expunge',
+        remote_side='Department.id',
+        single_parent=True)
+
+    @classmethod
+    def to_id(cls, department):
+        if not department:
+            return None
+
+        if isinstance(department, six.string_types):
+            try:
+                department = int(department)
+            except ValueError:
+                return department
+
+        if isinstance(department, int):
+            # This is the same algorithm used by the migration script to
+            # convert c.JOB_LOCATIONS into department ids in the database.
+            prefix = '{:07x}'.format(department)
+            return prefix + str(uuid.uuid5(cls.NAMESPACE, str(department)))[7:]
+        else:
```

Comment:
Minor style nitpick: The if/else pairing here doesn't make total sense, since functionally this is a giant if/elif/elif/else statement but you've chosen to rely on `return` statements to break the loop. This pattern is totally fine, but when I do it I always take the final 'return' statement and put it outside of all the if statements, at the end. So it looks like:
```
if X:
  return A

if Y:
  return B

if Z:
  return C

return D
```

It's functionally the same as what you've done here, I just like it a little better that way.